### ISSUE TITLE
Normal loss, but very large valid WER

### ISSUE THREAD

**Issue by kaiyikang** [speechbrain/speechbrain#974]
Hi,

I want to replace the **seq2seq LibriSpeech recipe** (located at `speechbrain/recipes/LibriSpeech/ASR/seq2seq`) to **Tedlium2** dataset.

## Issue:
When training the first epoch, the **WER** of the validation dataset is huge. It looks like this:

![image](https://user-images.githubusercontent.com/22473900/132531939-cb353c0a-2e79-481e-892d-3c17e9b4d1ea.png)

You can see, the valid CER and valid WER are large. 
Is this case normal? As far as I know, WER or CER should not exceed 100%.

Thanks so much.

## Solved
After some epochs, it converges. Due to the too-simple LM, the first test WER is so bad. And I set the lm_weight = 0, then the result is much better. Unfortunately, I don't have a way to get/train the best LM, so I will decode without LM.

![image](https://user-images.githubusercontent.com/22473900/132944416-9b7fa6f4-50e7-45f8-a5f5-f5a45f26a1f6.png)

## More Details
Due to the difference between the Librispeech and Tedlium2, I created a new tokenizer and trained a new LM model.

### 1. Tokenizer
I write a script to generate {train, dev, test}.json files. It looks like:

![image](https://user-images.githubusercontent.com/22473900/132528918-c32837e7-964c-431f-a6c1-ddd5e4a72706.png)

The config is unchanged, and I get the **1000_unigram.model** and **1000_unigram.vocab**. it looks like this:
~~~
<unk>	0
S	-2.87499
▁THE	-3.45784
▁AND	-3.82364
▁A	-3.84872
▁TO	-3.95718
T	-4.02102
▁	-4.07242
▁OF	-4.0772
'	-4.11417
▁THAT	-4.30122
▁I	-4.30406
~~~

### 2. LM 
I used configuration **RNNLM.yaml** to train the LM.
The input text looks like this:

![image](https://user-images.githubusercontent.com/22473900/132529890-9d3cc7df-1a75-468b-a5ec-fabe95f77b40.png)

The tokenizer is generated from the last step and the config of LM model is default:

~~~
model: !new:speechbrain.lobes.models.RNNLM.RNNLM
   output_neurons: !ref 1000
   embedding_dim: 128
   activation: !name:torch.nn.LeakyReLU
   dropout: 0.0
   rnn_layers: 2
   rnn_neurons: 2048
   dnn_blocks: 1
   dnn_neurons: 512
~~~

### 3. ASR
For ASR, I only change the audio processing pipeline in **train.py** and I also created CSV files.

**audio pipeline**
~~~python
    # 2. Define audio pipeline:
    @sb.utils.data_pipeline.takes("wav","stime","etime",)
    @sb.utils.data_pipeline.provides("sig")
    def audio_pipeline(file_path, start_time, end_time):
        sample_rate = 16000
        start_time = int(float(start_time) * sample_rate)
        end_time = int(float(end_time) * sample_rate)
        length = end_time - start_time
        backend = torchaudio.get_audio_backend()
        if backend == "sox" or (backend == "soundfile" and torchaudio.USE_SOUNDFILE_LEGACY_INTERFACE):
            kwargs = {"offset": start_time, "num_frames": length}
        else:
            kwargs = {"frame_offset": start_time, "num_frames": length}
        sig,sr = torchaudio.load(file_path, **kwargs)
        yield sig.transpose(0, 1).squeeze(1)
    sb.dataio.dataset.add_dynamic_item(datasets, audio_pipeline)
~~~

**CSV file**
~~~
ID,duration,wav,spk_id,wrd,stime,etime
AlGore_2009-1,10.42,/home/lujun/asr/kaldi/egs/tedlium/s5_r2/db/TEDLIUM_release2/dev/sph/AlGore_2009.sph,Al_Gore,LAST YEAR I SHOWED THESE TWO SLIDES SO THAT DEMONSTRATE THAT THE ARCTIC ICE CAP WHICH FOR MOST OF THE LAST THREE MILLION YEARS HAS BEEN THE SIZE OF THE LOWER FORTY EIGHT STATES HAS SHRUNK BY FORTY PERCENT,13.04,23.46
AlGore_2009-18,4.39,/kaldi/egs/tedlium/s5_r2/db/TEDLIUM_release2/dev/sph/AlGore_2009.sph,Al_Gore,WHAT THIS IS DOING,205.16,209.55
AlGore_2009-34,1.41,/kaldi/egs/tedlium/s5_r2/db/TEDLIUM_release2/dev/sph/AlGore_2009.sph,S45,SMELLS GOOD TOO,343.36,344.77
~~~

In config file, I make the size of encoder smaller:
~~~
activation: !name:torch.nn.LeakyReLU
dropout: 0.15
cnn_blocks: 2
cnn_channels: (128, 256)
inter_layer_pooling_size: (2, 2)
cnn_kernelsize: (3, 3)
time_pooling_size: 4
rnn_class: !name:speechbrain.nnet.RNN.LSTM
rnn_layers: 4
rnn_neurons: 512
rnn_bidirectional: True
dnn_blocks: 2
dnn_neurons: 512
emb_size: 128
dec_neurons: 512
output_neurons: 1000  # Number of tokens (same as LM)
blank_index: 0
bos_index: 0
eos_index: 0
unk_index: 0
~~~

I can make sure that the tokenizer and LM loading is correct. The training process is also no problem.

**Comment by mravanelli** [comment]
In the first few epochs this is normal. Then you should quickly converge.
Please, let us know!

**Comment by kaiyikang** [comment]
Thanks so much, it converges. I have updated the result.

I have tried to fine-tune the LM based on `speechbrain/asr-crdnn-rnnlm-librispeech` and Ted-lium2 dataset (lm_weight=0.5),  but the WER(`wer=30.16%`) is higher than lm_weight = 0(`wer=15.95%`). 

I don't have any idea how to make the LM better. So currently, I will decode without LM.

**Comment by Gastron** [comment]
Great, looks like this was solved!
